apps/homework/tasks.py

```diff
@@ -14,41 +15,63 @@
 
 logger = logging.getLogger(__name__)
 
+
 class SubmissionPostException(BaseException):
 
     def __init__(self, *args, **kwargs):
         self.sub_type = kwargs.get('sub_type')
         super(SubmissionPostException, self).__init__(*args, **kwargs)
 
+
+def retrieve_score_from_jupyter_notebook(fd):
+    content = fd.readlines()
+    filtered_content = list(filter(lambda line: 'final score' in str(line) and 'print' not in str(line), content))
+    if len(filtered_content) is not 1:
+        raise IndexError('Exactly one line should contain the phrase "final score" and not include the phrase "print".')
+    score_string = str(filtered_content[0])
+    start_index = score_string.find('is')
+
+    found = re.search('[0-9]+ \/ [0-9]+', score_string)
```

Comment:
like we talked about, could add the full string here for better context as to what this is doing.

Also we should have tests using a dummy ipynb and such (you were probably planning on this already)

Reply:
✅